### DISCUSSION TITLE
ETHGlobal Brussels Hackathon Guide by MetaMask + Linea

### DISCUSSION THREAD

**Discussion by Montoya** [MetaMask/snaps#2471] [Announcements]
Unlock secure, seamless blockchain experiences with MetaMask and Linea, your gateway to dapp innovation.

We are offering a $5000 bounty for dapps deployed on Linea and seamlessly integrated into MetaMask with the MetaMask SDK and MetaMask Snaps ($2500 for 1st place, $1500 for 2nd place, and $1000 for 3rd place).

Examples projects: 
- Build a dapp on Linea and use the MetaMask SDK for React to seamlessly onboard users in your frontend
- Build a MetaMask Snap that reads from and interacts with your smart contract on Linea (like the [Linea Voyager Snap](https://github.com/Consensys/linea-voyager-snap))

Documentation and resources to help you on your journey: 

* [Workshop demo that has MetaMask SDK + wagmi already wired up and deploys to Linea](https://github.com/Consensys/web3-fullstack-starter)
* [CLI to scaffold a fullstack monorepo that uses either NextJS or ViteJS + Hardhat or Foundry to get started, and also includes viem and wagmi with a Connect button that uses MetaMask SDK](https://github.com/Consensys/create-web3-template)
* [Linea Docs](https://docs.linea.build/developers/hackathons)
* [Linea Tutorials](https://docs.linea.build/developers/guides/community)
* [Linea Quickstart](https://docs.linea.build/developers/quickstart)
* [MetaMask SDK](https://docs.metamask.io/wallet/how-to/use-sdk/) (you must use an SDK library in your dapp frontend, `window.ethereum` will not qualify!)
* Connector libraries that use MetaMask SDK: 
  * [Wagmi](https://docs.metamask.io/wallet/how-to/use-sdk/3rd-party-libraries/wagmi/)
  * [Web3Onboard](https://docs.metamask.io/wallet/how-to/use-sdk/3rd-party-libraries/web3-onboard/)
* [MetaMask Snaps Docs](https://docs.metamask.io/snaps/)

**Comment by NPCoc** [comment]
gg

**Comment by sasim081** [comment]
very good
